website/src/plugins/github-release-plugin.ts

```diff
@@ -0,0 +1,206 @@
+import type { Plugin } from '@docusaurus/types';
+
+// Type definition for an asset from the GitHub API response
+interface GitHubAsset {
+  name: string;
+  browser_download_url: string;
+}
+
+// Type definition for the relevant parts of the GitHub release API response
+interface GitHubRelease {
+  tag_name?: string;
+  name: string;
+  assets: GitHubAsset[];
+}
+
+// --- Data type definitions for each OS ---
+
+export interface LinuxDownloadData {
+  version: string;
+  flatpak: string;
+  amd64: string;
+  arm64: string;
+}
+
+export interface MacosDownloadData {
+  version: string;
+  universal: string;
+  x64: string;
+  arm64: string;
+  airgapsetupX64: string;
+  airgapsetupArm64: string;
+}
+
+export interface WindowsDownloadData {
+  version: string;
+  binaryX64: string;
+  binaryArm64: string;
+  setupX64: string;
+  setupArm64: string;
+  airgapsetupX64: string;
+  airgapsetupArm64: string;
+}
+
+// --- Combined data types ---
+
+export interface AllDownloadsData {
+  linuxDownloads: LinuxDownloadData;
+  macosDownloads: MacosDownloadData;
+  windowsDownloads: WindowsDownloadData;
+}
+
+export interface GlobalData {
+  linuxDownloads: LinuxDownloadData;
+  macosDownloads: MacosDownloadData;
+  windowsDownloads: WindowsDownloadData;
+}
+
+// Helper function to find an asset or throw an error
+function findAssetOrThrow(
+  assets: GitHubAsset[],
+  predicate: (asset: GitHubAsset) => boolean,
+  assetName: string,
+): string {
+  const asset = assets.find(predicate);
+  if (!asset) {
+    throw new Error(`Required asset not found: ${assetName}`);
+  }
+  return asset.browser_download_url;
+}
+
+// The main plugin function, now strongly typed
+export default async function githubReleasePlugin(): Promise<Plugin<AllDownloadsData | null>> {
+  return {
+    name: 'docusaurus-plugin-github-release',
+
+    async loadContent(): Promise<AllDownloadsData | null> {
+      console.log('Fetching latest GitHub release for podman-desktop...');
+
+      try {
+        const response = await fetch('https://api.github.com/repos/containers/podman-desktop/releases/latest');
+        if (!response.ok) {
+          throw new Error(`Failed to fetch latest release: ${response.statusText}`);
+        }
+        const releaseData = (await response.json()) as GitHubRelease;
```

Comment:
Wondering about the Octokit usage instead of raw request to the API directly

Reply:
581c8c4e1b671546e37a18b8fde4555be6354e0f